### PR TITLE
i18n(de) Translate de/guides/deploy/gitlab.md

### PR DESCRIPTION
**Pull request by tobiasschmidt89**
Translate de/guides/deploy/gitlab.md

#### What kind of changes does this PR include?

- Translated content

#### Description

- Translate de/guides/deploy/gitlab.md
- Visual test with pnpm dev
- Site is confusing written in EN and has issues, aligned with doc team to leaf it like that for now

### CODE REVIEW

File: src/pages/de/guides/deploy/gitlab.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+title: Veröffentliche deine Astro-Website mit GitLab Pages
+description: Anleitung zur Veröffentlichung deiner Astro-Website mittels GitLab Pages.
+layout: ~/layouts/DeployGuideLayout.astro
+---
+
+[GitLab Pages](https://pages.gitlab.io/) ermöglichen es dir eine Astro-Website für deine [GitLab](https://about.gitlab.com/) Projekte, Gruppen oder Benutzerkonten bereitzustellen.
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/gitlab.md:7]:
```suggestion
Du kannst [GitLab Pages](https://pages.gitlab.io/) verwenden, um eine Astro-Website für Projekte, Gruppen oder Benutzerkonten auf [GitLab](https://about.gitlab.com/) bereitzustellen.
```

File: src/pages/de/guides/deploy/gitlab.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+:::tip[Auf der Suche nach einem Beispiel?]
+Schau dir [das offizielle Astro-Beispiel für GitLab Pages an](https://gitlab.com/pages/astro)!
+:::
+
+## So funktioniert die Veröffentlichung
+
+1. Setze den Parameter `.site` in `astro.config.mjs` auf den benötigten Wert.
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/gitlab.md:15]:
```suggestion
1. Setze die Konfigurations&shy;option `.site` in `astro.config.mjs` auf den benötigten Wert.
```

File: src/pages/de/guides/deploy/gitlab.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+2. Setze den Parameter `dist` in `astro.config.mjs` auf den Wert `public` und den Parameter `public` in `astro.config.mjs` auf einen neu benannten Ordner, der alles enthält, was sich derzeit in `public` befindet. Der Grund dafür ist, dass `public` ein zweiter Quellordner in Astro ist. Damit du Dateien von `public` veröffentlichen kannst, musst du die entsprechenden Dateien aus einem anderen Ordner holen. Ihre "astro.config.mjs" könnte damit so aussehen:
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/gitlab.md:16]:
```suggestion
2. Setze die Option `dist` in `astro.config.mjs` auf den Wert `public` und die Option `public` in `astro.config.mjs` auf einen neu benannten Ordner, der alles enthält, was sich derzeit in `public` befindet. Der Grund dafür ist, dass `public` standardmäßig ein Quellverzeichnis für öffentliche Dateien in Astro darstellt. Damit du `public` stattdessen als Ausgabeverzeichnis verwenden kannst, musst du die öffentlichen Dateien aus einem anderen Ordner holen. Deine `astro.config.mjs` könnte damit so aussehen:
```

File: src/pages/de/guides/deploy/gitlab.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+   ```js
+   export default defineConfig({
+     sitemap: true,
+     site: 'https://astro.build/',
+   });
+   ```
+
+3. Erstelle eine Datei namens `.gitlab-ci.yml` im Stammverzeichnis deines Projekts mit folgendem Inhalt. Sobald du Änderungen an den Inhalten deiner Seite durchführst wird deine Website damit neu gebaut und veröffentlicht:
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/gitlab.md:25]:
```suggestion
3. Erstelle eine Datei namens `.gitlab-ci.yml` im Stammverzeichnis deines Projekts mit folgendem Inhalt. Sobald du Änderungen an den Inhalten deiner Seite durchführst, wird deine Website damit neu gebaut und veröffentlicht:
```

File: src/pages/de/guides/deploy/gitlab.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+## So funktioniert die Veröffentlichung
+
+1. Setze den Parameter `.site` in `astro.config.mjs` auf den benötigten Wert.
+2. Setze den Parameter `dist` in `astro.config.mjs` auf den Wert `public` und den Parameter `public` in `astro.config.mjs` auf einen neu benannten Ordner, der alles enthält, was sich derzeit in `public` befindet. Der Grund dafür ist, dass `public` ein zweiter Quellordner in Astro ist. Damit du Dateien von `public` veröffentlichen kannst, musst du die entsprechenden Dateien aus einem anderen Ordner holen. Ihre "astro.config.mjs" könnte damit so aussehen:
+
+   ```js
+   export default defineConfig({
+     sitemap: true,
+     site: 'https://astro.build/',
+   });
```
Comment (hippotastic) [inline src/pages/de/guides/deploy/gitlab.md:22]: Oh my. This code block doesn't contain any of the options discussed above. But this issue is also present in the English source page, so I'm not suggesting to change this right now.

**Review by hippotastic — COMMENTED** [review]
Good translation! Please find some minor suggestions below.

The source page has some issues that you already discovered, but fixing these is out of scope for this translation, so we should be good to go after the minor fixes. :)

File: src/pages/de/guides/deploy/gitlab.md (hunk 3 above)
Comment (hippotastic) [reply src/pages/de/guides/deploy/gitlab.md:16]: Ich habe hier etwas freier übersetzt, weil man den Originaltext ja schon kaum versteht. Die Dopplungen hinsichtlich des Konfigurationsdateinamens habe ich mal dringelassen, war aber schwer am Zögern, ob ich sie nicht auch streichen soll. :)

**Comment by tobiasschmidt89** [comment]
Ok good to merge then. Thanks.

**Pull request merged**